src/utils/event-type-guards.ts

```diff
@@ -58,3 +59,7 @@ export function isSNSEvent(event: any): event is SNSEvent {
 export function isSQSEvent(event: any): event is SQSEvent {
   return Array.isArray(event.Records) && event.Records.length > 0 && event.Records[0].eventSource === "aws:sqs";
 }
+
+export function isAppSyncResolverEvent(event: any): event is AppSyncResolverEvent<any> {
```

Comment:
This check may be a bit permissive? Can we be sure that event.info.variables is used only for AppSync related events? According to the doc there is a `selectionSetGraphQL` field in the event, this might be more tightly coupled with AppSync?

Reply:
:+1: I can switch to that